File: Seccion 12 - Punteros en C++/Seccion_12_Python/06_ejercicio_06.py

nElementos: int
elementos: list = []

def pedir_datos():
    global nElementos, elementos
    
    nElementos = int(input('Digite el número de elementos: '))
    
    for i in range(nElementos):
        val = int(input('Digite un número[{}]: '.format(i)))
        elementos.append(val)

def buscar_elemento():
    dato: int
    i: int
    band: bool = False
    
    dato = int(input('Digite un número a buscar: '))
    
    i = 0
    while i < nElementos and band == False:
        if elementos[i] == dato:
            band = True
        
        i += 1
        
    if band == False:
        print('\nEl numero no se encuentra en el arreglo.')
    else:
        print('\nEl número sí se encuentra en el arreglo, posición: {}'.format(i - 1))

def main():
    
    pedir_datos()
    
    buscar_elemento()

if __name__ == '__main__':
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/06_ejercicio_06.dart

import 'dart:io';

int nElementos;
List<int> elementos = new List<int>();

void pedirDatos() {
  stdout.write('Digite el número de elementos: ');
  nElementos = int.parse(stdin.readLineSync());

  for (int i = 0; i < nElementos; i++) {
    stdout.write('Digite un número[$i]: ');
    elementos.insert(i, int.parse(stdin.readLineSync()));
  }
}

void buscarElemento() {
  int dato, i;
  bool band = false;

  stdout.write('Digite un número a buscar: ');
  dato = int.parse(stdin.readLineSync());

  i = 0;
  while ((i < nElementos) && (band == false)) {
    if (elementos[i] == dato) {
      band = true;
    }

    i++;
  }

  if (band == false) {
    print('\nEl número no existe en el arreglo.');
  } else {
    print('\nEl número sí existe en el arreglo. posición: ${i - 1}');
  }
}

void main() {

  pedirDatos();

  buscarElemento();

}